lib/connection.js

```diff
@@ -563,7 +565,7 @@ Connection.prototype.retrieve = function(type, ids, options, callback) {
     _.map(ids, function(id) {
       if (!id) { return Promise.reject(new Error('Invalid record ID. Specify valid record ID value')).thenCall(callback); }
       var url = [ self._baseUrl(), "sobjects", type, id ].join('/');
-      return self.request(url);
+      return self.request(url, options);
```

Comment:
Here I pass in 'options' to request so the headers in the options can be applied to the API call. Note 'url' here is a string, not an object.

I can also reconstruct the request as : 
return self.request({ method: 'GET', url: url, headers:  _.defaults(options.headers || {})});
But I'd like to keep info in 'options' encapsulated and let request to handle 'options', just in case there is any info other than headers can be used in request. If in the future we want to extend the usage of 'options' in request, the way I am using here will make it more flexible.